docs/src/real_world.md

```diff
@@ -397,3 +397,25 @@ res = collect_results!(datadir("results"); black_list = bl, subfolders = true)
 select!(res, Not(:path)) # don't show path this time
 ```
 All `missing` entries were adjusted automatically :)
+
+## Defining parameter sets with restrictions
+
+As already demonstrated in the examples above, for functions where the set of input parameters is the same for each simulation run, a basic dictionary can be used to define these parameters.
+However, often some of the parameters or values should only be considered if another parameter is also included in the set or has a specific value.
+The macro [`@onlyif`](@ref) allows to place such restrictions on values and parameters.
+The following dictionary defines values and parameters for a genetic algorithm:
+
+```julia
+ga_parameters = Dict(
+    :population_size => [20,50,100],
+    :selection => ["roulette-selection", "SUS", "tournament-selection", "linear ranking"],
+    :fitness_scaling => @onlyif(:selection in ("SUS", "roulette-selection"), collect(1.0:20.0)),
+    :tournamet_size => @onlyif(:selection == "tournament-selection", collect(2:10)),
+    :chromosome => [:A, @onlyif(begin
+        size_constr = :population_size <= 50
+        select_constr = :selection != "SUS"
+        size_constr && select_constr
```

Comment:
`size_constr && select_constr` what does this line do?

Reply:
AHA! It says "only if condition `size_constr` and `select_constr` are met! Maybe you add a comment to this after the exectution for clarity

Reply:
Damn, gotta say, this is so pretty advanced stuff :D

Reply:
Usually I would put this on one line, but I wanted to show, that one can use `begin end`.
Both variables are boolean

Reply:
only if both conditions are met, return true?

Reply:
yeah right. One could reason (to have a real world example), that this type of chromosome has a bad performance for a population size larger than 50 and one needs to use a fast selection algorithm ...